### DISCUSSION TITLE
Increase maximum subscription limit = 100  (ignores those above it)

### DISCUSSION THREAD

**Discussion by KenSense** [FreeOpcUa/opcua-asyncio#1174] [Q&A]
Hi,

Using SINUMERIK OPC UA Server not sure who set the 100 subscriptions.
Those above 100 are ignored,  `await sub.subscribe_data_change(subscribe_node_list)`
If I make two lists the limit is the same, it stops at 100 subscriptions on the server.

I need 150 subscriptions, is that server side or is it somewhere in opcua-asyncio?

Thanks

**Comment by AndreasHeine** [comment]
150 subscriptions or 150 monitored items ?

a subscription is just a long-polling-like transport mechanism which can have thousands of monitored items (nodes/variables)

![image](https://user-images.githubusercontent.com/56362817/211881409-b91cc920-f58e-423d-b70c-06e11a77ed34.png)

**Reply by AndreasHeine** [reply]
there is also a limit on how many monitored items you can add in one call!

please check the server operational limits:
![image](https://user-images.githubusercontent.com/56362817/211882611-e2161002-2f85-4134-b861-59ef909bb494.png)

example subscribe maxpercall aware:
https://github.com/AndreasHeine/opcua-tutorial/blob/main/client/client1.py

**Comment by KenSense** [comment]
I have 144 nodes I subscribe to, many are arrays:
Like 1 subscription/node is "ns=2;s=/Channel/Parameter/R[u1, 1, 50]" gives me a notification if 1 of those 50 variables changes.

I checked the serverCapabilities but mine doesn't seem to mention a MaxSubscriptions.
So should I assume there isn't one on the server or is it just not provided?
![image](https://user-images.githubusercontent.com/49442805/212021782-014bfa04-02c7-4645-af9e-30240db36840.png)

String/byte sizes are all very big 16.7MB, MaxArraylength=65536  so I don't see a problem there.

edit:
I do get an error in UaExpert when trying to go over this 100:
![image](https://user-images.githubusercontent.com/49442805/212049103-cffd7087-ef96-4758-94af-1cf8481b4a9a.png)

edit2: removed some to get below 100, since it's probably a hard server limit.
